### ISSUE TITLE
Support .editorconfig's end_of_line option in the editor

### ISSUE THREAD

**Issue by silverwind** [gogits/gogs#3550]
The web editor should set the line ending output format of the editor according to the `end_of_line` option if it's present:

> end_of_line: set to "lf", "cr", or "crlf" to control how line breaks are represented.

Spawned off https://github.com/gogits/gogs/issues/3516.

**Comment by andreynering** [comment]
Agreed. Just keep in mind that `lf` is almost always the right setting. On Windows, Git converts from `crlf` to `lf` (and the other way around on checkout) by default to avoid the big mess of having mixed line endings. `lf` should be the absolute default.

**Comment by jcgruenhage** [comment]
@andreynering at the moment, this is the default setting, a week ago or so `crlf` was the default. Do you that it should be just kept that way and `crlf` files that are for example uploaded should be automatically changed to `lf` as well? I think we should just leave the files as they are :D

**Comment by andreynering** [comment]
@jcgruenhage I haven't thought of uploaded files, but I think they should be converted to LF, unless they are not text files (binary) or .editorconfig specifies CRLF.

**Comment by silverwind** [comment]
I still think we should preserve original line endings by detecting them (e.g. https://github.com/gogits/gogs/issues/3516#issuecomment-242785686 and the following comment).

Of course, a `.editorconfig` setting overrules that detection.
